### ISSUE TITLE
Notices related to the publishing workflow should be announced to assistive technologies

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#3690]
Any visual notice used in the editing workflow should be announced with an audible message to screen readers. For example, when publishing a post a notice appears:

![screen shot 2017-11-28 at 13 16 48](https://user-images.githubusercontent.com/1682452/33319518-4c0c8a16-d43f-11e7-9e01-47b986ce2e9a.png)

Since there's no page reload any longer (the page reload was sort of reassuring feedback that something happened), screen reader users won't have a clue of what happened. Was the post published / updated successfully? Any errors? All they get is just silence.

All these kinds of visual messages should always be paired with an audible message to convey the same information to users.

**Comment by mtias** [comment]
Also related to: https://github.com/WordPress/gutenberg/issues/3496

I think this should all be done together, and using the speak accessibility utility makes a lot of sense to me.
